src/diffusers/loaders/lora.py

```diff
@@ -1033,17 +1036,73 @@ def enable_lora_for_text_encoder(self, text_encoder: Optional["PreTrainedModel"]
     def set_adapters(
         self,
         adapter_names: Union[List[str], str],
-        adapter_weights: Optional[List[float]] = None,
+        adapter_weights: Optional[Union[List[float], List[Dict]]] = None,
```

Comment:
Can the type annotation be more precise than `Dict`? I think it's `Dict[str, Union[float, int]]`?